src/createSignRequest.js

```diff
@@ -0,0 +1,59 @@
+'use strict'
+const constructSignatureString = require('./constructSignatureString')
+const constructDigestString = require('./constructDigestString')
+
+const createSignRequest = (creationOptions) => {
```

Comment:
Are we sure we need a creator function? None of the `creationOptions` are used outside of the return function.

Reply:
This was a suggestion from @jkirkpatrick24 to allow the signing options to be set once and then have a reduced number of parameters for each request signed.

Reply:
I think abstractions like that are best left up to the user. One could always wrap the function in a callback and apply their params, but we shouldn't assume everyone would prefer that.
```js
const signRequest = require('@autotelic/ietf-hmac-signatures')

module.exports = (req) => {
  return signRequest(req, {
    signedHeaders: ['request-target', 'created', 'content-length', 'custom-header'],
  })
}
```

Reply:
This is a good point but I think it requires more discussion before we set on a path. Is this something we can handle in an issue and as part of the discussion over @jkirkpatrick24 's spike?

We could get an initial release completed as part of this sprint and then make heavy modifications later.